hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalMigration.java

```diff
@@ -0,0 +1,293 @@
+    }
+
+    @Override
+    public String getDescription() {
+        return "Update the is_approval value for synthetic crypto transfers";
+    }
+
+    @Override
+    protected MigrationVersion getMinimumVersion() {
+        // The version where the nft_transfer table was migrated to the transaction table
+        return MigrationVersion.fromVersion("1.81.0");
+    }
+
+    @Override
+    protected Long getInitial() {
+        return LOWER_BOUND_TIMESTAMP;
+    }
+
+    @Override
+    protected int getSuccessChecksum() {
+        return 1;
+    }
+
+    @Override
+    protected Optional<Long> migratePartial(Long lastUpperBound) {
+        if (!MirrorProperties.HederaNetwork.MAINNET.equalsIgnoreCase(mirrorProperties.getNetwork())) {
+            log.info("Skipping migration since it only applies to mainnet");
+            return Optional.empty();
+        }
+
+        AtomicLong count = new AtomicLong(0L);
+        var migrationErrors = new ArrayList<String>();
+        long upperBound = getUpperBound(lastUpperBound);
+        Map<String, Long> boundaryTimestamps = Map.of("lower_bound", lastUpperBound, "upper_bound", upperBound);
+        try {
+            var transfers = transferJdbcTemplate.query(TRANSFER_SQL, boundaryTimestamps, transferMapper);
+            for (ApprovalTransfer transfer : transfers) {
+                if (!isAuthorizedByContractKey(transfer, migrationErrors)) {
+                    // set is_approval to true
+                    String updateSql = UPDATE_TOKEN_TRANSFER_SQL;
+                    if (transfer.transferType == TRANSFER_TYPE.CRYPTO_TRANSFER) {
+                        updateSql = UPDATE_CRYPTO_TRANSFER_SQL;
+                    } else if (transfer.transferType == TRANSFER_TYPE.NFT_TRANSFER) {
+                        updateSql = UPDATE_NFT_TRANSFER_SQL;
+                    }
+
+                    var updateMap = Map.of(
+                            "amount", transfer.amount,
+                            "consensus_timestamp", transfer.consensusTimestamp,
+                            "index", transfer.index,
+                            "sender", transfer.sender,
+                            "token_id", transfer.tokenId);
+                    transferJdbcTemplate.update(updateSql, updateMap);
+                    count.incrementAndGet();
+                }
+            }
+        } catch (Exception e) {
+            log.error("Error migrating transfer approvals", e);
+            return Optional.empty();
+        }
+
+        log.info("Updated {} crypto transfer approvals", count);
+        migrationErrors.forEach(log::error);
+        return upperBound == UPPER_BOUND_TIMESTAMP ? Optional.empty() : Optional.of(upperBound);
+    }
+
+    private Long getUpperBound(long lastUpperBound) {
+        long nextIncrement = Math.addExact(lastUpperBound, TIMESTAMP_INCREMENT);
+        return nextIncrement >= UPPER_BOUND_TIMESTAMP ? UPPER_BOUND_TIMESTAMP : nextIncrement;
+    }
```

Comment:
there is no concern of overflow, simplify it, or perhaps just remove the function since it's a one-liner now

```suggestion
    private Long getUpperBound(long lowerBound) {
        return Math.min(lowerBound + TIMESTAMP_INCREMENT, UPPER_BOUND_TIMESTAMP);
    }
```

Reply:
I've updated it and removed the method.